### DISCUSSION TITLE
Dialog translation support / String consistency

### DISCUSSION THREAD

**Discussion by sairuk** [retronas/discussions#110] [Ideas]
Future state thought

Refactor the dialogs to source a strings file at the top of each script that will not only provide string consistency, reducing duplication but will also permit sourcing translations based on the $LANG or similar env vars. 

something like 

```
_CONFIG=/opt/retronas/dialog/retronas.cfg
_LANG=/opt/retronas/etc/lang/$LANG.string

rn_tcpser() {
source $_CONFIG
source $_LANG
dialog \
  --backtitle "${RN_TITLE}" \
  --title "${RN_TITLE} ${RN_TCPSER} ${RN_MENU}" \
  --clear \
  --menu "${RN_IPADDRESS}: ${MY_IPS} \
  \n
  \n${RN_PLSOPT}" ${MG} 10 \
  "01" "${RN_MAIN_MENU}" \
```

where 

`/opt/retronas/etc/lang/$LANG.string` would look like

```
RN_TITLE=RetroNAS
RN_TCPSER=TCPSer
RN_MENU=menu
RN_IPADDRESS="My IP addresses:"
RN_PLSOPT=Please select an option
RN_MAIN_MENU=Main Menu
```

**Comment by danmons** [comment]
Very high on my want list.  Being English-only is a real problem I feel. 

Goes for the Wiki too.

**Comment by FlotterCodername** [comment]
Once this is in, ping me for a German translation.
